### ISSUE TITLE
Wrong number created from constructor

### ISSUE THREAD

**Issue by elvisun** [noojee/money.dart#26]
When rounding with decimals some math is wrong, for example, this is true when it should be false:

```dart
      Money.from(69.99, Currency.create('USD', 2)).minorUnits.toInt() == 6998; // Wrong, should be 6999
```
But this doesn't happen on all integers:

```dart
      Money.from(29.99, Currency.create('USD', 2)).minorUnits.toInt() == 6999; // This is correct
```

**Comment by elvisun** [comment]
On plugin version `money2: ^1.4.3`

Flutter version:
```
[√] Flutter (Channel stable, 1.22.0, on Microsoft Windows [Version 10.0.19041.685], locale en-US)
    • Flutter version 1.22.0 at D:\flutter
    • Framework revision d408d302e2 (4 months ago), 2020-09-29 11:49:17 -0700
    • Engine revision 5babba6c4d
    • Dart version 2.10.0


[√] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at C:\Users\1\AppData\Local\Android\sdk
    • Platform android-30, build-tools 29.0.2
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[√] Android Studio (version 4.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 50.0.1
    • Dart plugin version 193.7547
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] IntelliJ IDEA Community Edition (version 2017.2)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2017.2.3
    • Flutter plugin version 21.2.2
    • Dart plugin version 172.3968.27

[√] Connected device (1 available)
    • Pixel 3a (mobile) • 04FAY1W0RT • android-arm64 • Android 10 (API 29)
```

**Comment by bsutton** [comment]
Fixed in:  2.0.1-nullsafety.7

**Comment by elvisun** [comment]
Thanks Brett, do you know when it'll make it into a stable release?
